### ISSUE TITLE
download youtube playlist item titles as batch

### ISSUE THREAD

**Issue by eflister** [ytdl-org/youtube-dl#30978]
- [x] I'm reporting a site feature request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've searched the bugtracker for similar site feature requests including closed ones

## Description

i need a list of titles in a youtube playlist as `youtube-dl` will save them.  right now, calling `youtube-dl` with a url like:
`> youtube-dl --verbose -i --restrict-filenames --get-filename https://www.youtube.com/playlist?list=<listName>`
seems to contact youtube individually for each item and is very slow for sizable playlists.

but the youtube api lets you get item details in batches and is very fast:
https://developers.google.com/youtube/v3/docs/playlistItems/list

also, if any items have non-`public` status (`unlisted`, `private`, or `unspecified` (deleted)), or are blocked by country or forbidden, i get errors like this:

```
WARNING: Unable to download webpage: <urlopen error [Errno 54] Connection reset by peer>
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/youtube_dl/extractor/youtube.py", line 1733, in _real_extract
    self._sort_formats(formats)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 1374, in _sort_formats
    raise ExtractorError('No video formats found')
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

**Comment by dirkf** [comment]
In general, each item title may not be set (or may not be set correctly) until the respective item has been resolved. Use `--flat-playlist` manually if you know that the extraction for the site does provide accurate titles in the playlist without fully extracting each video.

**Comment by eflister** [comment]
before closing this i think you should address the `<urlopen error [Errno 54] Connection reset by peer>` error and the name/help text for `--flat-playlist`.

**Comment by dirkf** [comment]
I think the Errno 54 is an artefact of some network device between you and YT as it's not normally seen.

The option name `flat-playlist` is traditional and the help text describes what it does, though I agree that it needs quite a good or even deep understanding of yt-dl to understand the description.

Someone who had time could write up a blog entry or a gist describing this.

**Comment by eflister** [comment]
i think ok to keep the current name, but what do you think of adding a synonym like `get-titles-only` and switching to the above proposed help text?

how do i debug the errno 54?  my hardware is just simplest possible residential wifi router from comcast, no weird settings.  you don't get the error if you try to run on a youtube playlist with items that have non-public status (unlisted, private, or unspecified (deleted)), or are blocked by country or forbidden?  (no error when using `--flat-playlist`)

**Comment by dirkf** [comment]
As I mentioned, the option doesn't actually "get-titles-only", it just doesn't run the extraction for each playlist item. The detail in the playlist items depends on how the playlist was created. If the playlist webpage that was downloaded or the API in use lists details for each item, the extractor might add those to the playlist, making them available at the `--flat-playlist` stage; otherwise the playlist may have, at a minimum, just the URLs of the playlist items. Therefore the solution to your original problem depends on which site is involved. (That's probably the blog post I was thinking of.)

To debug Errno 54, find a page that provokes the error. If you can, try accessing it from a different LAN, coffeeshop or whatever; also, try setting up a free VPN and accessing it that way.

Basically, if YT is giving you the bum's rush, you normally get an application-level code, like 401, 403, 429. A TCP reset would be done by something that can't send an application-level error because it can't see into the secure application-level exchange.
